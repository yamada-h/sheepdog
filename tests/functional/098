#!/bin/bash

# Test recovery process of erasure coding in a case of nr nodes < nr data stripe

. ./common

for i in 0 1 2; do
    _start_sheep $i
done
_wait_for_sheep 3
_cluster_format -c 2:1

$DOG vdi create test 64M

_random | $DOG vdi write test 0 $((4 * 1024 * 1024 * 64))

$DOG node list

BEFORE=`$DOG vdi read test | md5sum`

_kill_sheep 1
_wait_for_sheep_stop 1
$DOG cluster info | _filter_cluster_info

_kill_sheep 2
_wait_for_sheep_stop 2
$DOG cluster info | _filter_cluster_info

for i in 1 2; do
    _start_sheep $i
done
_wait_for_sheep 3
$DOG cluster info | _filter_cluster_info

$DOG node list

AFTER=`$DOG vdi read test | md5sum`

if [[ $BEFORE = $AFTER ]]
then
    echo "checksum matched, success"
else
    echo "checksum not matched, fail"
    echo "before: " $BEFORE
    echo "after:  " $AFTER
fi
