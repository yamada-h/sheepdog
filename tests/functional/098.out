QA output created by 098
using backend plain store
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      2 [127.0.0.1:7000, 127.0.0.1:7002]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      3 [127.0.0.1:7000]
DATE      2 [127.0.0.1:7000, 127.0.0.1:7002]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      5 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
DATE      4 [127.0.0.1:7000, 127.0.0.1:7001]
DATE      3 [127.0.0.1:7000]
DATE      2 [127.0.0.1:7000, 127.0.0.1:7002]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
checksum matched, success
